Still crash on load due to missing image file

@mpaperno  @LorenzMeier
     I have a problem in anting the project jmavsim.  I also get a crash on boot because the image is not found:

Exception in thread "main" java.lang.IllegalArgumentException: input == null!
    at javax.imageio.ImageIO.read(ImageIO.java:1388)
    at me.drton.jmavsim.Visualizer3D$CustomCanvas3D.<init>(Visualizer3D.java:707)
    at me.drton.jmavsim.Visualizer3D.<init>(Visualizer3D.java:114)
    at me.drton.jmavsim.Simulator.<init>(Simulator.java:209)
    at me.drton.jmavsim.Simulator.main(Simulator.java:611)
I also used followed code, but it still report this errors. 
  java -Djava.ext.dirs= -cp lib/*:out/production/jmavsim.jar me.drton.jmavsim.Simulator


In aternate build method. the terminal report  that the error is followed:

- BUILD FAILED
Target "creat_run_jar" does not exist in the project "jMAVSim". 

OS:Ubuntu 14.04 64bit
FIle:/home/user

```
BUILD FAILED Target "creat_run_jar" does not exist in the project "jMAVSim". 
                         ^
```
Please check your spelling.

I don't know why the images don't work when trying to start it the "old way."  It shouldlook in `environment/` folder relative to the executing directory.  

Someone could try to debug the line throwing the error -- perhaps the way the image is being loaded (the classLoader) is causing the problem.  There are quite a few ways to load a URL/path in Java... one of them should work (as an example, there's another way used in `KinematicObject.modelFromFile()`).

```java
// load and scale compass image for overlay
Image img = ImageIO.read(getClass().getClassLoader().getResource(TEX_DIR + COMPASS_IMG));
```

-Max

--- a/src/me/drton/jmavsim/Visualizer3D.java
+++ b/src/me/drton/jmavsim/Visualizer3D.java
@@ -711,15 +711,15 @@ public class Visualizer3D extends JFrame {
             
             try {
                 file = new URL("file:./" + TEX_DIR + COMPASS_IMG);
-            } catch (MalformedURLException e) {
+                if (file != null) {
+                    Image img = ImageIO.read(file);
+                    img = img.getScaledInstance(overlayWidth, overlayWidth, Image.SCALE_SMOOTH);
+                    compassOverlay.createGraphics().drawImage(img,  0,  0, null);
+                }
+                
+            } catch (IOException e) {
                 System.err.println("Error, could not load image: " + TEX_DIR + COMPASS_IMG);
                 System.err.println(e);
-            }
-            
-            if (file != null) {
-                Image img = ImageIO.read(file);
-                img = img.getScaledInstance(overlayWidth, overlayWidth, Image.SCALE_SMOOTH);
-                compassOverlay.createGraphics().drawImage(img,  0,  0, null);
             }
             
             // set up vector lines for HUD

--- a/src/me/drton/jmavsim/Visualizer3D.java
+++ b/src/me/drton/jmavsim/Visualizer3D.java
@@ -17,9 +17,11 @@ import java.awt.geom.AffineTransform;
 import java.awt.geom.Line2D;
 import java.awt.geom.RoundRectangle2D;
 import java.awt.image.BufferedImage;
-import java.io.BufferedOutputStream;
+//import java.io.BufferedOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.util.List;
 import java.util.ArrayList;
 import java.util.BitSet;
@@ -289,7 +291,7 @@ public class Visualizer3D extends JFrame {
         if (texture == null)
             System.err.println("Cannot load texture from " + fn);
         else {
-            System.out.println("Loaded texture from " + fn);
+            //System.out.println("Loaded texture from " + fn);
             texture.setEnable(true);
         }
         return texture;
@@ -704,10 +706,21 @@ public class Visualizer3D extends JFrame {
             drawg2d = drawImg.createGraphics();
 
             // load and scale compass image for overlay
-            Image img = ImageIO.read(getClass().getClassLoader().getResource(TEX_DIR + COMPASS_IMG));
-            img = img.getScaledInstance(overlayWidth, overlayWidth, Image.SCALE_SMOOTH);
+            URL file = null;
             compassOverlay = new BufferedImage(overlayWidth, overlayWidth, BufferedImage.TYPE_4BYTE_ABGR);
-            compassOverlay.createGraphics().drawImage(img,  0,  0, null);
+            
+            try {
+                file = new URL("file:./" + TEX_DIR + COMPASS_IMG);
+            } catch (MalformedURLException e) {
+                System.err.println("Error, could not load image: " + TEX_DIR + COMPASS_IMG);
+                System.err.println(e);
+            }
+            
+            if (file != null) {
+                Image img = ImageIO.read(file);
+                img = img.getScaledInstance(overlayWidth, overlayWidth, Image.SCALE_SMOOTH);
+                compassOverlay.createGraphics().drawImage(img,  0,  0, null);
+            }
             
             // set up vector lines for HUD
             hdgLine = new Line2D.Float(0, 0, 0, halfW * -0.85f);

